### ISSUE TITLE
Strange/bad cycling results

### ISSUE THREAD

**Issue by vectro** [GIScience/openrouteservice#212]
Regarding [this route](https://maps.openrouteservice.org/directions?n1=40.760606&n2=-73.964226&n3=17&a=40.748965,-73.912779,40.765235,-73.982263&b=1e&c=0&g1=2&g2=7&h2=3&k1=en-US&k2=km), ORS is suggesting to take 61st Street to Lexington Avenue (to 55th Street), but neither of these ways have any cycle facilities. Meanwhile it is possible to take 1st Avenue (Southbound) to 59th Street to 2nd Avenue, which provides 100% coverage of cycle facilities with no obvious difference in elevation profile.

**Comment by simple-php** [comment]
I'm having the [same kind of problems](https://maps.openrouteservice.org/directions?n1=50.705679&n2=5.817985&n3=17&a=50.704497,5.820436,50.706834,5.816585&b=1c&c=0&g1=-1&g2=0&h2=3&k1=en-US&k2=km). Instead of following the road around the corner, it takes some steps and a small footway/path while the profile is set to **cycling-road**.

**Comment by marq24** [comment]
just for clarification - is 'this' the route that you are looking for?
https://maps.openrouteservice.org/directions?n1=40.759932&n2=-73.966522&n3=17&a=40.748965,-73.912779,40.760366,-73.963952,40.765235,-73.982263&b=1e&c=0&g1=2&g2=7&h2=3&k1=en-US&k2=km

It would be cool if when you have an issue with routing, that you would provide the shortest version that still having the issue - this helps (at least me) to concentrate on the real issue (TIA)

**Comment by vectro** [comment]
[shortest version with issue](https://maps.openrouteservice.org/directions?n1=40.757709&n2=-73.973866&n3=15&a=40.760253,-73.961903,40.765235,-73.982263&b=1e&c=0&g1=2&g2=7&h2=3&k1=en-US&k2=km)
The best route I can get out of ORS is [this](https://maps.openrouteservice.org/directions?n1=40.764911&n2=-73.981951&n3=17&a=40.748965,-73.912779,40.760366,-73.963952,40.763918,-73.980474,40.765235,-73.982263&b=1e&c=0&g1=2&g2=7&h2=3&k1=en-US&k2=km), but it's still suboptimal; better would be to just turn right on Broadway and walk the half block on the sidewalk.

**Comment by marq24** [comment]
Allow me to differentiate between some aspects I have understood so far... and I will try to comment/explain...

Let me start with a general hint - one thing you can always check is the openstreetmap section of the area in question - since all we see in ors is based on that data...
https://www.openstreetmap.org/edit#map=20/40.76484/-73.98186

That's how it started for me btw - I had also issues with e bike route and realized that the source data in osm was wired - so after I have started to corrected the osm data I got better results for my bike routing.

So coming back to Broadway & 55'th Street - from the "TurnRestrictions" and "onway" attributes I would not see, that it should be possible to make there a right hand turn with a bike. 

Of course I can understand as human, that you might do that turn anyway and then get off the bike and walk the few meters - but this human nature [to switch your 'vehicle-type' (from bike to foot)] will currently not be considered... (when I say considered, then I mean "calculated").

Please don't get me wrong - I just want to explain you what currently is implemented - and that based on the current information available in osm source data for a bike there is no option to offer the "human-natural route"... 

But before you are getting frustrated - I am a bike rider myself - so in the past few days I have started the development of an alternative bike routing - started as personal project - now part of the openrouteservice - I can't promise you any release dates - but I can ensure you that I am highly motivated to improve the output of ors. Have said that, I still need to lower your expectation for this Broadway & 55'th Street right turn... I'll have to check the new code with the NY area - but I doubt that we will get this "fixed" (walk the last few meters) in a short period of time...

For the other mentioned problematic area - from the bridge, onto the 1'st Av - and then to 59'th I am some sort of confident that this might be already working with the changes implemented (but need to be verified)

**Comment by vectro** [comment]
Marq24, this makes sense. I'm less worried about the right turn onto Broadway, which, as you note, requires a mode change. The thing that concerns me is the use of 61st Street and of Lexington Ave., when 55th Street is available. Let me know if I can do anything more to help with that.

**Comment by marq24** [comment]
... just downloaded nyc osm data and processing the data - I'll keep you updated...

just to double check - this rout would be fine - right?
https://maps.openrouteservice.org/directions?n1=40.762479&n2=-73.970207&n3=16&a=40.758619,-73.958416,40.76447,-73.981934&b=1d&c=0&g1=2&g2=7&h2=3&k1=en-US&k2=km

**Comment by vectro** [comment]
Yes, that is a good route, assuming the user would then walk to the final destination. Thanks.

**Comment by marq24** [comment]
just a short update - also the TouringBike profile will return a "valid" route... 

https://maps.openrouteservice.org/directions?n1=40.755287&n2=-73.974681&n3=14&a=40.748965,-73.912779,40.764624,-73.982019&b=1d&c=0&g1=2&g2=7&h2=3&k1=en-US&k2=km

It look like that regular & safety cycling profiles focus more on the classification of the road itself - while the other profiles (touring or mtb) prefer a route with separated cycle line... 

Right now my tests with the new road bike profile returned a total different route - mainly via the "Central Park South"... for sure not the way you wanted to go.

I would kindly ask for some patience till we have resolved internally some things here in our development (we just have moved to a never version of graphhopper) and we want to make small steps...

So let's see how the result will be once the new ors (based on gh 0.10) will look like - and what else can/need to be adjusted in order to get better routing results (in urban areas). It's quite a challenge toe get comparable results on country side (that's where I live & test) and urban areas (which are still familiar to me - but that's a different story) ;-)

**Comment by vectro** [comment]
Makes sense to me. Ironic that the "touring" option makes better use of the bike lanes than the "safety" option, since 55th Street is definitely safer than 57th Street, but also slower (at least on weekdays).

**Comment by marq24** [comment]
might be just too many options (aka profiles) with to less differences... and the main driver of the default routing is "speed"... 

As I already said - we want to optimize the bike routing in general (and might get rid of some options in order to reduce complexity)

**Comment by brmnr** [comment]
I didn't know whether to start a new issue, or using this issue about some strange cycling results I'm having as well. I'm playing around with ORS and "cycling-road" directions for some time now, but frequently i get strange cycling issue like [this one](https://maps.openrouteservice.org/directions?n1=50.817866&n2=5.877031&n3=16&a=50.814151,5.88879,50.819344,5.870454&b=1a&c=0&g1=-1&g2=0&h2=3&k1=en-US&k2=km). While it should stay on the "Ingbergrachtweg" and "Ingberdorpstraat"  (which are perfect for road cycling), it takes some strange side-ways and I cannot see why?

**Comment by marq24** [comment]
when you toggle between 'fastest' & 'shortest' you can see (in your given example) that the code calculates a ride time of 11min for the "natural preferred" direct line - in contrast to 9min with the two additional loops. 

I know this is not a very smart answer - but by toggle between fastest & shortest you might can get at least an explanation "why" you see a certain route...

The "how fast can I ride on a specific section" calculation is currently under investigation - and you example is once again an indication that the current implementation need to be reviewed (which is exactly what I currently do).

**Comment by rabidllama** [comment]
@vectro @marq24 @rbremen I have just tried these with the new version of ORS (currently live) and I am just trying to organise things.
In terms of the original issue in New York, using regular bike now takes you south on 1st Avenue, and then west on East 55th Street. There is a bit where it goes east first on the East 59th, but I am not sure if that is because of 1 way systems. 
![image](https://user-images.githubusercontent.com/7126643/47083693-b3588500-d211-11e8-8a9f-c65515b2210d.png)
So basically, is this more like what the route should be like (note that we no longer have a "safe" cycling profile active)?

As for the second issue relating to the use of side roads, it seems that still occurs so I will create a new issue for that which can then be addressed rather than trying to address two things in one issue :)

**Comment by vectro** [comment]
This route is a little better than what was showing previously, but I think it's still suboptimal. I don't see any advantage to deviating to Sutton Place when it would be possible to take 59th Street directly to 2nd Avenue.  In other words, it would be better to turn right on 59th Street, rather than turning left.

**Comment by rabidllama** [comment]
After looking into the other issue aswell, it seems that the results for the new york on is down to elevation... From what I can work out the system must pick up a longer section of incline in the "best" route and so sends it on a detour. We are looking into ways of updating the evaluation of how changes of elevation affect travel speen, so hopefully that will make its way into the code in the next week

**Comment by vectro** [comment]
That makes sense to me. Personally I would view total climbing as a better metric than incline, but others might disagree. I did some quick Internet searches but didn't immediately find any relevant research.

**Comment by rabidllama** [comment]
I have closed the #327 issue so that we can keep all the discussion here to reduce duplication. It definitely seems to be something relating to the use of elevation in speed calculations (see my comments on the #327 issue), so we can schedule this for the 4.7.2 release which gives us time to look into alternatives

**Comment by rabidllama** [comment]
We will be rebuilding graphs in the coming weeks so hopefully that will address some of the issues we see in relation to the elevation that gets used for speed calculations

**Comment by rabidllama** [comment]
We have turned off the elevation effects in the 4.7.2 release so it should now work more like expected. Fell free to let us know if it is still doing some massively weird things.

**Comment by brmnr** [comment]
Although it works a lot better now,  I'm still getting weird results on some roads where it's making very strange detours. For example this road: http://maps.openrouteservice.org/directions?n1=50.808531&n2=5.929109&n3=16&a=50.808484,5.919753,50.806179,5.924013&b=1c&c=0&g1=-1&g2=0&h2=31&k1=en-US&k2=km
While it should stay on the same straight road from A to B (Partijerweg), it takes a very odd detour. That road is perfectly suitable for cycling and not closed or somehow.

**Comment by rabidllama** [comment]
I will create a new issue for this as this doesn't seem to be related to what was causing the previous problem
